etna/transforms/math/sklearn.py

```diff
@@ -227,8 +227,8 @@ def _inverse_transform(self, df: pd.DataFrame) -> pd.DataFrame:
 
     def _preprocess_macro(self, df: pd.DataFrame) -> np.ndarray:
         segments = sorted(set(df.columns.get_level_values("segment")))
-        x = df.loc[:, pd.IndexSlice[:, self.in_column]]
-        x = pd.concat([x[segment] for segment in segments]).values
+        x = df.loc[:, pd.IndexSlice[segments, self.in_column]]
```

Comment:
I guess we don't need this line here as it is part of `to_flatten` with `features` argument setted